### PR TITLE
Update cpython test suite configuration

### PR DESCRIPTION
**Pull request by bodzhang**
- Enable more test cases
  - Seperate cpython3.8 and cpython3.9 test configuration, as certain cpython3.9
    test cases are not present in cpython3.8
  - Add config.json file to set current working directory in Mystikos environment
  - Add more depedency modules in the Dockerfile

Signed-off-by: Bo Zhang (ACC) <zhanb@microsoft.com>

### CODE REVIEW

**Comment by bodzhang** [comment]
cpython test passed in [this](https://openenclave.visualstudio.com/ACC-Services/_build/results?buildId=10927&view=results) nightly pipeline run

File: tests/cpython-tests/Makefile [hunk 1]
```diff
@@ -2,18 +2,18 @@ TOP=$(abspath ../..)
 include $(TOP)/defs.mak
 
 APPBUILDER=$(TOP)/scripts/appbuilder
-HEAP_SIZE="4G"
-OPTS += --memory-size $(HEAP_SIZE) --nobrk
+
+OPTS = --nobrk --app-config-path config.json
```
Comment (vtikoo) [inline tests/cpython-tests/Makefile:6]: nit: --nobrk can also be specified in config.json

**Review by vtikoo — APPROVED** [review]

**Comment by vtikoo** [comment]
We should reevaluate at some point if splitting of the passing tests list is still required. Originally I had split the test lists due to memory leaks.

File: tests/cpython-tests/Makefile (hunk 1 above)
Comment (bodzhang) [reply tests/cpython-tests/Makefile]: Added "NoBrk" in config.json and keep command line options for Linux target

File: tests/cpython-tests/Makefile [hunk 2]
```diff
@@ -2,18 +2,23 @@ TOP=$(abspath ../..)
...
-OPTS += --memory-size $(HEAP_SIZE) --nobrk
+
+OPTS = --app-config-path config.json
 ifdef STRACE
 OPTS += --strace
 endif
+
+#Linux target ignores config.json, add equivalent command line options as a workaround
+HEAP_SIZE="4G"
+OPTS += --memory-size $(HEAP_SIZE) --nobrk
```
Comment (vtikoo) [inline tests/cpython-tests/Makefile:13]: This can be wrapped in a `ifeq ($TARGET, linux)`.

Comment (bodzhang) [reply tests/cpython-tests/Makefile]: In fact, Linux target does process config.json, but have slightly different logic than sgx target. Luckily, for this specific config.json, the difference in the logic does not have negative impact. Removed the command line options in the latest update to the PR.

**Pull request merged**